### ISSUE TITLE
Cannot appear the given metrics

### ISSUE THREAD

**Issue by zscwind** [Wenjun-Peng/fastformer-for-rec#4]
<html>
<body>

models | AUC | MRR | nDCG@5 | nDCG@10
-- | -- | -- | -- | --
official | 0.7268 | 0.3745 | 0.4151 | 0.4684
ours | 0.7256 | 0.3720 | 0.4101 | 0.4660
mine | 0.7115 | 0.3491 | 0.3882 | 0.4506

</body>
</html>

Dear Author，I do the experiment again, but cannot appear the given metrics(auc=0.7256)，when my auc is only 0.7115. 
Compare with your parameters：
`python train.py \`
`--pretreained_model unilm \`
`--pretrained_model_path {path to ckpt of unilmv2} \`
`--root_data_dir ./data/speedy_data/ \`
`--num_hidden_layers 8 \`
`--world_size 4 \`
`--lr 1e-4 \`
`--pretrain_lr 8e-6 \`
`--warmup True \`
`--schedule_step 240000 \`
`--warmup_step 1000 \`
`--batch_size 42 \`
`--npratio 4 \`
`--beta_for_cache 0.002 \`
`--max_step_in_cache 2 \`
`--savename speedyrec_mind \`
`--news_dim 256`
I set `world_size=1`，and the pretrained model is `unilm2-base-uncased`. Except this，other parameters are the same with above. Could you help me find the auc gap? Thank you very much!

**Comment by Wenjun-Peng** [comment]
You can download the pretrained model given by microsoft, and finetune with my model. [https://drive.google.com/drive/folders/1Aw9Rgc9gyr_3eRU6_cksxq1uiEe7LYGb](url)

**Comment by zscwind** [comment]
> You can download the pretrained model given by microsoft, and finetune with my model. [link](https://drive.google.com/drive/folders/1Aw9Rgc9gyr_3eRU6_cksxq1uiEe7LYGb)

Thank you. By the way, the model in link is unilm2 ? or trained by you ?

**Comment by zscwind** [comment]
> You can download the pretrained model given by microsoft, and finetune with my model. [link](https://drive.google.com/drive/folders/1Aw9Rgc9gyr_3eRU6_cksxq1uiEe7LYGb)

Thanks. By the way, the model (auc=0.7256) is trained with which pretrained model? unilm2 or the model in link?

**Comment by Wenjun-Peng** [comment]
The model (auc=0.7256) is trained with [link](https://drive.google.com/drive/folders/1Aw9Rgc9gyr_3eRU6_cksxq1uiEe7LYGb)

**Comment by zscwind** [comment]
> The model (auc=0.7256) is trained with [link](https://drive.google.com/drive/folders/1Aw9Rgc9gyr_3eRU6_cksxq1uiEe7LYGb)

Let me see. The model [link](https://drive.google.com/drive/folders/1Aw9Rgc9gyr_3eRU6_cksxq1uiEe7LYGb) is based by [speedy_mind](https://github.com/microsoft/SpeedyRec/blob/main/speedy_mind/README.md) with the pretained model unilm2, right?

**Comment by Wenjun-Peng** [comment]
Yes, actually, the training pipeline is based on the speedy_mind




------------------&nbsp;Original message&nbsp;------------------
From: "zscwind"; 
Sendtime: Monday, Jan 9, 2023 6:57 PM
To: "Wenjun-Peng/fastformer-for-rec"; 
Cc: ***@***.***>; "Comment"; 
Subject: Re: [Wenjun-Peng/fastformer-for-rec] Cannot appear the given metrics (Issue #4)





  
The model (auc=0.7256) is trained with link
  
Let me see. The model link is based by speedy_mind with the pretained model unilm2, right?

**Comment by zscwind** [comment]
> Yes, actually, the training pipeline is based on the speedy_mind
Speedy_mind is really a wonderful framework. But there is still 2 problems confused me:
1. On my machines, the prefetch can't work properly. So I have to set `prefetch=false`
2. The task is blocked on multi gpus mode(world_size>1), the I set `world_size=1`
Did you meet the same problems? Your experience may be helpful, thanks

**Comment by zscwind** [comment]
> The model (auc=0.7256) is trained with [link](https://drive.google.com/drive/folders/1Aw9Rgc9gyr_3eRU6_cksxq1uiEe7LYGb)

I am training the model with  [link](https://drive.google.com/drive/folders/1Aw9Rgc9gyr_3eRU6_cksxq1uiEe7LYGb)，unexpectedly the traning auc is very poor. I guess the parameter setting is wrong. Can you share your training parameters?

**Comment by Yaobink** [comment]
公开的代码是不是没有集成训练的部分呢

**Comment by Wenjun-Peng** [comment]
> > The model (auc=0.7256) is trained with [link](https://drive.google.com/drive/folders/1Aw9Rgc9gyr_3eRU6_cksxq1uiEe7LYGb)
> 
> I am training the model with [link](https://drive.google.com/drive/folders/1Aw9Rgc9gyr_3eRU6_cksxq1uiEe7LYGb)，unexpectedly the traning auc is very poor. I guess the parameter setting is wrong. Can you share your training parameters?

The hyperparams shown in the readme are exact what my training used

**Comment by Wenjun-Peng** [comment]
> 公开的代码是不是没有集成训练的部分呢

这个代码包含了训练预测和提交，但是是非官方，我也只是根据论文的描述，尽我所能训练出靠近论文指标的模型
